### ISSUE TITLE
[Bug] App crash when clicking notification

### ISSUE THREAD

**Issue by nronaldvdberg** [firebase/quickstart-unity#1070]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2020.3.7f1
  * Firebase Unity SDK version: 7.2.0
  * Source you installed the SDK: .unitypackage
  * Problematic Firebase Component: Messaging
  * Other Firebase Components in use: Analytics
  * Additional SDKs you are using: Admob
  * Platform you are using the Unity editor on: Windows
  * Platform you are targeting: Android
  * Scripting Runtime: IL2CPP

### [REQUIRED] Please describe the issue here:

When app is running, it will crash when user taps a push notification from the app in the notification tray. (When the app is not open, behavior is normal: tapping a push notification will open the app).

Everything was running fine before. I believe this started after I removed and then re-imported Firebase SDK to my unity project. That may have updated the SDK version (i'm not sure which one i was using before the re-import).

Logs:

0001/01/01 00:00:00.000 -1 -1 Info  --------- beginning of crash
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime FATAL EXCEPTION: main
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime Process: com.xxx.yyy, PID: 9663
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime java.lang.IllegalArgumentException: Scheduled service ComponentInfo{com.xxx.yyy/com.google.firebase.messaging.MessageForwardingService} does not require android.permission.BIND_JOB_SERVICE permission
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.os.Parcel.createExceptionOrNull(Parcel.java:2389)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.os.Parcel.createException(Parcel.java:2369)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.os.Parcel.readException(Parcel.java:2352)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.os.Parcel.readException(Parcel.java:2294)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.job.IJobScheduler$Stub$Proxy.enqueue(IJobScheduler.java:342)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.JobSchedulerImpl.enqueue(JobSchedulerImpl.java:56)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at androidx.core.app.JobIntentService$JobWorkEnqueuer.enqueueWork(JobIntentService.java:343)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at androidx.core.app.JobIntentService.enqueueWork(JobIntentService.java:523)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at androidx.core.app.JobIntentService.enqueueWork(JobIntentService.java:501)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at com.google.firebase.messaging.MessageForwardingService.enqueueWork(MessageForwardingService.java:37)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at com.google.firebase.MessagingUnityPlayerActivity.onNewIntent(MessagingUnityPlayerActivity.java:75)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.Activity.performNewIntent(Activity.java:8213)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1409)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1422)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.ActivityThread.deliverNewIntents(ActivityThread.java:4031)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.ActivityThread.handleNewIntent(ActivityThread.java:4043)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.servertransaction.NewIntentItem.execute(NewIntentItem.java:53)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.os.Handler.dispatchMessage(Handler.java:106)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.os.Looper.loop(Looper.java:246)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.ActivityThread.main(ActivityThread.java:8512)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at java.lang.reflect.Method.invoke(Native Method)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime Caused by: android.os.RemoteException: Remote stack trace:
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at com.android.server.job.JobSchedulerService$JobSchedulerStub.enforceValidJobRequest(JobSchedulerService.java:2590)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at com.android.server.job.JobSchedulerService$JobSchedulerStub.enqueue(JobSchedulerService.java:2673)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.app.job.IJobScheduler$Stub.onTransact(IJobScheduler.java:175)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.os.Binder.execTransactInternal(Binder.java:1190)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 	at android.os.Binder.execTransact(Binder.java:1159)
2021/06/07 07:32:02.729 9663 9663 Error AndroidRuntime 

#### Relevant Code:

Here is AndroidManifest file in the Plugins/Android folder:

```
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" xmlns:tools="http://schemas.android.com/tools" package="com.xxx.yyy" android:versionCode="1" android:versionName="1.0">
  <application android:label="@string/app_name" android:icon="@drawable/app_icon" android:theme="@style/UnityThemeSelector">
    <meta-data android:name="unityplayer.SkipPermissionsDialog" android:value="true" />  
    <!-- The MessagingUnityPlayerActivity is a class that extends UnityPlayerActivity to work around a known issue when receiving notification data payloads in the background. -->    
    <activity android:name="com.google.firebase.MessagingUnityPlayerActivity" android:configChanges="fontScale|keyboard|keyboardHidden|locale|mnc|mcc|navigation|orientation|screenLayout|screenSize|smallestScreenSize|uiMode|touchscreen">
      <intent-filter>
        <action android:name="android.intent.action.MAIN" />
        <category android:name="android.intent.category.LAUNCHER" />
      </intent-filter>
      <meta-data android:name="unityplayer.UnityActivity" android:value="true" />
    </activity>
    <service android:name="com.google.firebase.messaging.MessageForwardingService" android:exported="false" />
  </application>
</manifest>
```

**Comment by Str4tos** [comment]
Hello.
We get the same crash.
Only one version of `Android 10 (SDK 29)` is affected.
```
java.lang.Error: FATAL EXCEPTION [main]
Unity version     : 2018.4.26f1
Device model      : Xiaomi M2010J19SI
Device fingerprint: Redmi/lime_in/lime:10/QKQ1.200830.002/V12.0.7.0.QJQINXM:user/release-keys
Caused by
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4290)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4322)
  at android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:52)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2058)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:226)
  at android.app.ActivityThread.main (ActivityThread.java:7592)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:539)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:950)
Caused by: java.lang.IllegalArgumentException: 
  at android.os.Parcel.createException (Parcel.java:2078)
  at android.os.Parcel.readException (Parcel.java:2042)
  at android.os.Parcel.readException (Parcel.java:1990)
  at android.app.job.IJobScheduler$Stub$Proxy.enqueue (IJobScheduler.java:342)
  at android.app.JobSchedulerImpl.enqueue (JobSchedulerImpl.java:52)
  at androidx.core.app.JobIntentService$JobWorkEnqueuer.enqueueWork (JobIntentService.java:343)
  at androidx.core.app.JobIntentService.enqueueWork (JobIntentService.java:523)
  at androidx.core.app.JobIntentService.enqueueWork (JobIntentService.java:501)
  at com.google.firebase.messaging.MessageForwardingService.enqueueWork (MessageForwardingService.java:37)
  at com.google.firebase.MessagingUnityPlayerActivity.onNewIntent (MessagingUnityPlayerActivity.java:75)
  at android.app.Activity.performNewIntent (Activity.java:7935)
  at android.app.Instrumentation.callActivityOnNewIntent (Instrumentation.java:1407)
  at android.app.Instrumentation.callActivityOnNewIntent (Instrumentation.java:1420)
  at android.app.ActivityThread.deliverNewIntents (ActivityThread.java:3534)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4273)
Caused by: android.os.RemoteException: 
  at com.android.server.job.JobSchedulerService$JobSchedulerStub.enforceValidJobRequest (JobSchedulerService.java:2732)
  at com.android.server.job.JobSchedulerService$JobSchedulerStub.enqueue (JobSchedulerService.java:2815)
  at android.app.job.IJobScheduler$Stub.onTransact (IJobScheduler.java:175)
  at android.os.Binder.execTransactInternal (Binder.java:1024)
  at android.os.Binder.execTransact (Binder.java:997)
```

![image](https://user-images.githubusercontent.com/22005013/120998752-0c91d000-c791-11eb-8e62-e53819ca9a07.png)  


Thanks

**Comment by nanddo** [comment]
We're observing the same issue with the same stack trace here.

`java.lang.Error: FATAL EXCEPTION [main]
Unity version     : 2019.4.21f1
Device model      : samsung SM-G991U
Device fingerprint: samsung/o1qsqw/o1q:11/RP1A.200720.012/G991USQU3AUDB:user/release-keys
Caused by
  at android.os.Parcel.createExceptionOrNull (Parcel.java:2389)
  at android.os.Parcel.createException (Parcel.java:2369)
  at android.os.Parcel.readException (Parcel.java:2352)
  at android.os.Parcel.readException (Parcel.java:2294)
  at android.app.job.IJobScheduler$Stub$Proxy.enqueue (IJobScheduler.java:342)
  at android.app.JobSchedulerImpl.enqueue (JobSchedulerImpl.java:56)
  at androidx.core.app.JobIntentService$JobWorkEnqueuer.enqueueWork (JobIntentService.java:343)
  at androidx.core.app.JobIntentService.enqueueWork (JobIntentService.java:523)
  at androidx.core.app.JobIntentService.enqueueWork (JobIntentService.java:501)
  at com.google.firebase.messaging.MessageForwardingService.enqueueWork (MessageForwardingService.java:37)
  at com.google.firebase.MessagingUnityPlayerActivity.onNewIntent (MessagingUnityPlayerActivity.java:75)
  at android.app.Activity.performNewIntent (Activity.java:8213)
  at android.app.Instrumentation.callActivityOnNewIntent (Instrumentation.java:1409)
  at android.app.Instrumentation.callActivityOnNewIntent (Instrumentation.java:1422)
  at android.app.ActivityThread.deliverNewIntents (ActivityThread.java:4031)
  at android.app.ActivityThread.handleNewIntent (ActivityThread.java:4043)
  at android.app.servertransaction.NewIntentItem.execute (NewIntentItem.java:53)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2307)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8512)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
Caused by: android.os.RemoteException: 
  at com.android.server.job.JobSchedulerService$JobSchedulerStub.enforceValidJobRequest (JobSchedulerService.java:2590)
  at com.android.server.job.JobSchedulerService$JobSchedulerStub.enqueue (JobSchedulerService.java:2673)
  at android.app.job.IJobScheduler$Stub.onTransact (IJobScheduler.java:175)
  at android.os.Binder.execTransactInternal (Binder.java:1190)
  at android.os.Binder.execTransact (Binder.java:1159)`

**Comment by nanddo** [comment]
@nronaldvdberg @Str4tos I've found [this thread](https://stackoverflow.com/questions/47325618/) on stack overflow and it solved the issue in my case. We're replaced the following line on `AndroidManifest.xml`:

Before: ```<service android:name="com.google.firebase.messaging.MessageForwardingService" android:exported="false" />```
After...: ```<service android:name="com.google.firebase.messaging.MessageForwardingService" android:permission="android.permission.BIND_JOB_SERVICE" android:exported="true"/>```

This adds the `BIND_JOB_SERVICE` permission to the build and also sets the exported to `true`.

We haven't dig deeper into the solution to fully understand it, though. I'd recommend checking if this works for you too, at least as a temporary fix for the problem.

**Important**: the step to reproduce provided above works. On our case, we had two devices with different builds (before and after codes above) running in background (i.e. minimized) and we sent a push from Firebase Cloud Messaging. The device with "Before" code crashed when opening the notification, while the device with "After" code worked fine. In more details:
1. Open the app
2. Minimize it
3. Send a notification from Firebase Cloud Message
4. Tap on the notification
5. Notice that the app crashes if there's no `BIND_JOB_SERVICE` permission added.

Hope that helps!

**Comment by nronaldvdberg** [comment]
@nanddo Thanks a lot! That seems to have solved the issue for me as well. These manifest files are still a bit of a mystery to me, so i can't contribute much in terms of trying to understand what the issue really was and why this solves it, but i'm happy it works for now at least :)

**Comment by mgrogin** [comment]
@paulinon We're also seeing the same issue in production... Where in the documentation or in the release notes does it say to add android:permission="android.permission.BIND_JOB_SERVICE" android:exported="true"/> ?

thanks

**Comment by paulinon** [comment]
Hi @mgrogin,

It's possible that this bug has been fixed in the latest version of the SDK (8.0.0) without the help of the workaround. If the issue persists, you may file a new ticket regarding this.
